e2e/src/index.axios.spec.ts

```diff
@@ -230,7 +230,9 @@ describe.each(startServerFunctions)(
         expect(err.response?.data).toMatchObject({
           message: "Request validation failed parsing request header",
           phase: "request_validation",
-          cause: expect.stringContaining("Expected number, received nan"),
+          cause: expect.stringMatching(
+            /Expected number, received nan|Invalid input: expected number, received NaN/,
```

Comment:
bit dirty, but adjust the e2e error expectations to accept either the zod v3 or v4 messages, allow the same test suite to be run against both generations.